### ISSUE TITLE
improvement: allow manual passed key prop from `b-table-column` in `b-table`

### ISSUE THREAD

**Issue by frantic1048** [rafaelpimpa/buefy#238]
### Overview of the problem

**Buefy** version: 0.5.0
**Vuejs** version: 2.4.2
**OS/Browser**: Various
### Description

I am creating a `b-table` with conditional column, the `scopedSlots` prop of <b-table> is a function that returns different `b-table-column`s on different status.

e.g. show/hide or change columns on a specific condition.

While I found b-table component doesn't update properly when I am changing `scopedSlots ` prop passed to `b-table`.

### Steps to reproduce

1. set b-table's scopedSlots.default to a function `getSlots` that returns an array of `b-table-column`
2. make `getSlots` return 3 columns at condition A, and return 4 different columns at condition B.
3. change the condition for `getSlots`.

### Expected behavior

b-table should be updated to match correct columns returned by `getSlots` function

### Actual behavior

b-table is not being updated correctly.

### Possible reason and solution

I found `b-table` uses scopedSlots array's index as key, thus a key prop value is not always binded to the same b-table-column and table columns could not update properly by Vue.

A piece of related buefy source code: https://github.com/rafaelpimpa/buefy/blob/79f7119174946774926679b2ed87898375ecdc11/src/components/table/Table.vue#L44-L49

I am wondering if `t-table` could accept manual key prop from `b-table-column` component, and fallback to index when there is no key prop provided. If this is acceptable, I could make a quick PR for it.

**Comment by rafaelpimpa** [comment]
Can you make a repro for this? I'm working on something similar and want to test.

**Comment by frantic1048** [comment]
Thanks for the response. I'd make one on the weekend.

**Comment by frantic1048** [comment]
Here is the repo: https://github.com/frantic1048/buefy-issue-238

The symptom is similar to #239

**Comment by rafaelpimpa** [comment]
Just use the ``visible`` prop instead of removing or pushing to the slots prop:

```javascript
slots.push([
    <b-table-column visible={this.slotsType === 'default'} field="b" label="B" width="40">
        { props.row.b }
    </b-table-column>,
    <b-table-column visible={this.slotsType === 'default'} field="c" label="C" width="40">
        { props.row.c }
    </b-table-column>,
    <b-table-column visible={this.slotsType === 'default'} field="d" label="D" width="40">
        { props.row.d }
    </b-table-column>,
    <b-table-column visible={this.slotsType !== 'default'} field="e" label="E" width="40">
        { props.row.e }
    </b-table-column>
])
```

**Comment by frantic1048** [comment]
> Just use the visible prop instead of removing or pushing to the slots prop

I know this method. It helps when all columns are predictive, but not with dynamically generated ones (That's the case I concern about).

I just read the fix in 1eb35feb7d76715a82e3b557cb2dc2465ccf4a9b , it works.

Thanks for the sweet UI :)
